### PR TITLE
ocamlPackages.magic-trace: init at 1.1.0

### PR DESCRIPTION
**Pull request by Alizter**
###### Description of changes

We add magic-trace and it's dependency cohttp_static_handler. We also bump the version of owee.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by vbgl** [comment]
Something went wrong with the git history. Also this breaks `spacetime_lib`: maybe you can just mark that package as broken.

**Comment by ulrikstrid** [comment]
Result of `nixpkgs-review pr 222652` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)

**Comment by Alizter** [comment]
I will fix the history. I can attempt to fix that broken package too.

File: pkgs/development/ocaml-modules/spacetime_lib/default.nix [hunk 1]
```diff
@@ -6,7 +6,6 @@ lib.throwIfNot (lib.versionAtLeast "4.12" ocaml.version)
 buildDunePackage rec {
   pname = "spacetime_lib";
   version = "0.3.0";
-
   useDune2 = true;
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/spacetime_lib/default.nix:9]:
```suggestion
  duneVersion = "2";
```

File: pkgs/development/ocaml-modules/magic-trace/default.nix [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  pname = "magic-trace";
+  version = "v1.1.0";
+
+  minimalOCamlVersion = "4.12";
+  duneVersion = "3";
+
+  src = fetchFromGitHub {
+    owner = "janestreet";
+    repo = "magic-trace";
+    rev = "${version}";
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/magic-trace/default.nix:14]:
```suggestion
    rev = "v${version}";
```

File: pkgs/development/ocaml-modules/spacetime_lib/default.nix [hunk 3]
```diff
@@ -6,8 +6,7 @@ lib.throwIfNot (lib.versionAtLeast "4.12" ocaml.version)
 buildDunePackage rec {
   pname = "spacetime_lib";
   version = "0.3.0";
-
-  useDune2 = true;
+  duneVersion = "3";
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/spacetime_lib/default.nix:9]: Is this “3” intended or a typo?

**Pull request merged**